.github/workflows/run_functional_product_tests.yaml

```diff
@@ -10,12 +10,70 @@ on:
   workflow_dispatch:
 
 jobs:
-  product-tests:
-    name: Product Tests
+  # Build a matrix of product param combinations for testing.
+  build-matrix:
+    name: Build Matrix
     runs-on: windows-latest
     defaults:
       run:
         shell: powershell
+    steps:
+      - name: Checkout Repo
+        uses: actions/checkout@v4
+      - name: Generate Matrix
```

Comment:
A few thoughts:

1) At one point, I asked whether it was better to have code in the workflow file itself or in an external file.  The answer I received was "it doesn't matter."  Are we having second thoughts about this?  Are we looking to put all code in external files (including the smoke tests and publish workflow code)?  If we want to move this workflow to an external file, then I think we need to move all code.  And I think this could be done, though I think it should be its own issue (or maybe issues).

2) Do we have unit tests for workflows?  If so, please point me to them.  If not and we should, then like my last comment, I think we should this for all code.  If we want to start unit testing the workflows, I think this should be its own issue (or maybe issues).

3) The structure of the config data is documented in the workflow and there are copious comments that explain how they are parsed.  Moving this code to an external file would not change or improve this documentation.

4) We know the code is working simply because we can see a one-to-one correspondence between the testing params (which are documented in a canvas) and the workflow jobs